src/redux/filter/filter.js

```diff
@@ -0,0 +1,23 @@
+const SET_VISIBILITY_FILTER = 'SET_VISIBILITY_FILTER';
+const visibilityFilters = {
+  SHOW_ALL: 'SHOW_ALL',
+  SHOW_COMPLETED: 'SHOW_COMPLETED',
+  SHOW_ACTIVE: 'SHOW_ACTIVE',
+};
+
+const setVisibilityFilter = (filter) => ({
+  type: SET_VISIBILITY_FILTER,
+  filter,
+});
+
+const reducer = (state = visibilityFilters.SHOW_ALL, action) => {
+  switch (action.type) {
+    case SET_VISIBILITY_FILTER:
+      return action.filter;
+    default:
+      return state;
+  }
+};
+
+export { setVisibilityFilter };
+export default reducer;
```

Comment:
- [ ] I don't think you need this file/folder. For this project, all you need are the redux folders/files for books and categories. I think it would be nice to add features after the last milestone for this project. 👍

Reply:
@Gambit142, I think I had to create a separate PR for it as it is an additional requirement.
Thanks :pray: